docs/source/write-a-plugin.rst

```diff
@@ -30,3 +30,59 @@ your plugin's root directory:
 
 **Note**: Due to the way plugins are used by palm, you will need to re-install
 the plugin every time you want to test changes to the plugin
+
+Creating a Plugin Config
+========================
+
+If you wantto make your plugin configurable, you will need to create a plugin
+config object and provide it to the plugin constructor.
+
+The plugin config object:
+- must be a subclass of ``palm.plugins.base_plugin_config.BasePluginConfig``
+- must define a ``set`` method which returns a dictionary of config values
+- must be initialized with a model object, using the pydantic ``BaseModel``,
+  which defines the config schema and will be used to validate the config values.
+
+Here is an example of a plugin config object and how to use it in a plugin:
+
+.. code:: python
+
+  import click
+  from palm.plugins.base import BasePlugin
+  from palm.plugins.base_plugin_config import BasePluginConfig
+  from pydantic import BaseModel
+
+  # Define the config schema
+  class MyPluginConfigModel(BaseModel):
+      my_config_value: str
+
+  # Define the config object
+  class MyPluginConfig(BasePluginConfig):
+      def __init__(self):
+          super().__init__('my_plugin', MyPluginConfigModel)
+
+      def set(self):
+          return {
+              "my_config_value": click.prompt("Please enter a value for my_config_value", type=str)
+          }
```

Comment:
One internal debate I'm having is whether it's better to require `MyPluginConfigModel` as an argument in the initializer, or to define an `@abstractproperty` for `model` instead?

Curious whether you all prefer what's above, or this?

```python
class CoreConfig(BasePluginConfig):
    def __init__(self):
        super().__init__("core")

    @property
    def model(self) -> BaseModel:
        class ConfigModel(BaseModel):
            local_artifacts: str
            prod_artifacts: str

        return ConfigModel

    def set(self) -> dict:
        click.echo("set core plugin config")
        config = {
            'local_artifacts': click.prompt(
                "Where are your local artifacts?", type=str
            ),
            'prod_artifacts': click.prompt("Where are your prod artifacts?", type=str),
        }
        self.write(config)
        return config
```

Reply:
Yep I think I like current implementation better.